### PR TITLE
upgrade dependency aws-sdk-v2

### PR DESCRIPTION
**Pull request by plopezlpz**
*Issue #, if available:*

*Description of changes:*

Upgrade dependency to use aws-sdk-v2

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by harlow** [comment]
would be great to have a versioned directory! nice work

File: go/deaggregator/deaggregator.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ const (
 
 // DeaggregateRecords takes an array of Kinesis records and expands any Protobuf
 // records within that array, returning an array of all records
-func DeaggregateRecords(records []*kinesis.Record) ([]*kinesis.Record, error) {
+func DeaggregateRecords(records []*types.Record) ([]*types.Record, error) {
```
Comment (harlow) [inline go/deaggregator/deaggregator.go:25]: thoughts on updating this to use `[]types.Record`? The v2-sdk doesn't use pointers which means any consumer of this library is going to need to jump through some hoops to create a new slice of pointers to pass to this func

```
// Represents the output for GetRecords.
type GetRecordsOutput struct {

	// The data records retrieved from the shard.
	//
	// This member is required.
	Records []types.Record
```

File: go/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Go Kinesis Deaggregation Modules
+
+The Kinesis Deaggregation Modules for Go provide the ability to do in-memory deaggregation of standard Kinesis user records using the [Kinesis Aggregated Record Format](https://github.com/awslabs/amazon-kinesis-producer/blob/master/aggregation-format.md) to allow for more efficient transmission of records.
+
+There are 2 versions based upon the AWS SDK version you are using:
+
+| SDK | Project |
+| --- | ------- |
+|Version 0 | [v0](.) |
```
Comment (harlow) [inline go/README.md:9]: Is the first version referred to as v0 or v1? I'd assume v1

Comment (plopezlpz) [reply go/README.md:9]: when importing it it is `github.com/awslabs/kinesis-aggregation/go v0.0.0-20210630091500-54e17340d32f` so I left it as v0

Comment (harlow) [reply go/README.md:9]: Gotcha. so versioning it based on THIS repo vs the version of AWS SDK it's being consumed from

**Pull request merged**